server/portal/apps/workspace/api/views.py

```diff
@@ -214,25 +214,162 @@ def listing(self, client, request):
+                f"(archiveSystemDir like '%{qs_title}%') OR",
+                f"(appId like '%{query_string}%') OR",
+                f"(appId like '%{qs_lower}%') OR",
+                f"(appId like '%{qs_upper}%') OR",
+                f"(appId like '%{qs_title}%') OR",
+                f"(archiveSystemId like '%{query_string}%') OR",
+                f"(archiveSystemId like '%{qs_lower}%') OR",
+                f"(archiveSystemId like '%{qs_upper}%') OR",
+                f"(archiveSystemId like '%{qs_title}%'))",
+            ]
 
-        sql_queries = [
-            f"(tags IN ('portalName: {portal_name}')) AND",
-            f"((name like '%{query_string}%') OR",
-            f"(archiveSystemDir like '%{query_string}%') OR",
-            f"(appId like '%{query_string}%') OR",
-            f"(archiveSystemId like '%{query_string}%'))",
-        ]
+        # Keep the fill-to-limit logic ONLY for Finished/Failed status searches
+        is_finished = 'FINISHED' in status_searches
+        is_failed = 'FAILED' in status_searches
+
+        if is_finished or is_failed:
+            target_offset = offset
+            collected, skipped = [], 0
+            upstream_page_size = max(limit, 50)
+            upstream_offset = 0
+
+            while len(collected) < limit:
+                page = client.jobs.getJobSearchListByPostSqlStr(
+                    limit=upstream_page_size,
+                    startAfter=upstream_offset,
+                    orderBy='lastUpdated(desc),name(asc)',
+                    request_body={"search": sql_queries},
+                    select="allAttributes",
+                    headers={"X-Tapis-Tracking-ID": f"portals.{request.session.session_key}"}
+                )
+                if not page:
+                    break
+
+                upstream_offset += len(page)
+
+                for job in page:
+                    status = getattr(job, 'status', None)
+
+                    if is_finished:
+                        if status == 'FINISHED':
+                            pass
+                        elif status == 'FAILED':
+                            if not (has_timeout_message(job) and is_interactive(job)):
+                                continue
+                        else:
+                            continue
+
+                    elif is_failed:
+                        if status != 'FAILED':
+                            continue
+                        if has_timeout_message(job) and is_interactive(job):
+                            continue
+
+                    if skipped < target_offset:
+                        skipped += 1
+                        continue
+
+                    collected.append(job)
+                    if len(collected) == limit:
+                        break
+
+                if len(page) < upstream_page_size:
+                    break
+
+            return collected
 
         data = client.jobs.getJobSearchListByPostSqlStr(
```

Comment:
Use of `getJobSearchListByPostSqlStr` Tapis has a bug where I don't think they are using the limit and startAfter.  I see it on prod too so not just this branch.  If you expect more than 50 results for your search query, you can recreate it.  So for example, i have over 60 qgis jobs and then i can start scrolling and with pagination I can get 200 QGIS jobs but i it occur.  

@rstijerina , maybe we resolve this in a different jira issue.

Reply:
Reporting to CIC and tracking on our end via https://tacc-main.atlassian.net/browse/WP-1108